src/makeCalculator.js

```diff
@@ -4,7 +4,39 @@
  * @return {object}
  */
 function makeCalculator() {
-  // write code here
+  const calc = {
+    result: 0,
+
+    add(num) {
+      calc.result += num;
+    },
+
+    subtract(num) {
+      calc.result -= num;
+    },
+
+    multiply(num) {
+      calc.result *= num;
+    },
+
+    divide(num) {
+      calc.result /= num;
+    },
+
+    operate(callback, num) {
+      callback(num);
+
+      return calc;
+    },
+
+    reset() {
+      this.result = 0;
+
+      return calc;
```

Comment:
Similarly to the `operate` method, use `this` instead of `calc` in the `reset` method to refer to the current instance. This ensures that the method will work correctly even if it's called in a context that is not the original `calc` object.